src/main/java/com/android/volley/toolbox/Volley.java

```diff
@@ -86,8 +86,27 @@ public static RequestQueue newRequestQueue(Context context, HttpStack stack) {
     }
 
     private static RequestQueue newRequestQueue(Context context, Network network) {
-        File cacheDir = new File(context.getCacheDir(), DEFAULT_CACHE_DIR);
-        RequestQueue queue = new RequestQueue(new DiskBasedCache(cacheDir), network);
+        final Context appContext = context.getApplicationContext();
+        // Use a lazy supplier for the cache directory so that newRequestQueue() can be called on
+        // main thread without causing strict mode violation.
+        DiskBasedCache.FileSupplier cacheSupplier =
+                new DiskBasedCache.FileSupplier() {
```

Comment:
I don't think the implementation of this FileSupplier needs to support parallel access, since DiskBasedCache serializes access to the directory (or else it would have all sorts of bad races).

I'd probably just suggest you implement it as the simple:

```return new File(appContext.getCacheDir(),  DEFAULT_CACHE_DIR);```

rather than worrying about caching the File instance, since the performance hit should be negligible.

Reply:
Done.